src/en/comicfans/src/eu/kanade/tachiyomi/extension/en/comicfans/ComicFans.kt

```diff
@@ -0,0 +1,202 @@
+
+        return POST("$apiUrl/books/custom/MostPopularLocal#$page", apiHeaders, body)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val data = response.parseAs<PopularDto>().data
+        val hasNextPage = response.request.url.fragment!!.toInt() < data.totalPages
+
+        return MangasPage(data.list.map { it.toSManga(cdnUrl) }, hasNextPage)
+    }
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val mangaList = document.select(
+            "div:has(>.block-title-bar > .title:contains(New Updates))" +
+                "> .book-container > .book",
+        ).map { element ->
+            SManga.create().apply {
+                thumbnail_url = element.selectFirst("img")!!.attr("abs:src")
+                with(element.selectFirst(".book-name > a")!!) {
+                    title = text()
+                    setUrlWithoutDomain(attr("abs:href"))
+                }
+            }
+        }
+
+        return MangasPage(mangaList, false)
+    }
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/books".toHttpUrl().newBuilder().apply {
+            addQueryParameter("pageNumber", page.toString())
+            addQueryParameter("pageSize", "20")
+            fragment(page.toString())
+
+            if (query.isNotBlank()) {
+                addPathSegment("search")
+                addQueryParameter("keyWord", query)
+            } else {
+                filters.getUriPart<GenreFilter>()?.let {
+                    addQueryParameter("genre", it)
+                }
+                filters.getUriPart<LastUpdateFilter>()?.let {
+                    addQueryParameter("withinDay", it)
+                }
+                filters.getUriPart<StatusFilter>()?.let {
+                    addQueryParameter("status", it)
+                }
+            }
+        }.build()
+
+        val searchHeaders = apiHeadersBuilder()
+            .set("Accept", "*/*")
+            .build()
+
+        return GET(url, searchHeaders)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        popularMangaParse(response)
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("Text search ignores filters"),
+        Filter.Separator(),
+        GenreFilter(),
+        LastUpdateFilter(),
+        StatusFilter(),
+    )
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(response: Response): SManga {
```

Comment:
btw they have api endpoint for details too

`https://api.comicfans.io/comic-backend/api/v1/content/books/2179`
`$apiUrl/books/$id`